### PR TITLE
Upgrade to SPI 0.8, add missing interface methods

### LINKED ISSUE #58: Upgrade to R2DBC 0.8.0 M8
Changelog: https://r2dbc.io/spec/0.8.0.M8/CHANGELOG.txt
Javadoc: https://r2dbc.io/spec/0.8.0.M8/api/
Spec: https://r2dbc.io/spec/0.8.0.M8/spec/html/

### PR DESCRIPTION
**Pull request by elefeint**
The upgrade added a couple of trivial interface methods.

While I was there, I folded everything under a single for-loop in `SpannerRowMetadata`. It already had a for-loop and two stream pipeline. Since now there are three different pieces of data we need to keep track of, a single for-loop is simpler.

Fixes #58 .

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
-    <r2dbc.version>1.0.0.M7</r2dbc.version>
+    <r2dbc.version>0.8.0.M8</r2dbc.version>
```
Comment (meltsufin) [inline pom.xml]: On the surface, looks like a downgrade.

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRowMetadata.java [hunk 2]
```diff
@@ -32,27 +33,32 @@
 
   private final List<ColumnMetadata> columnMetadatas;
 
+  private final List<String> columnNames;
+
   /**
    * Mapping of column names to its integer index position in the row.
    */
-  private final HashMap<String, Integer> columnNameIndex;
+  private final HashMap<String, Integer> columnNameIndex = new HashMap<>();
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/spanner/r2dbc/SpannerRowMetadata.java]: What was wrong with initialization in the constructor?

File: pom.xml (hunk 1 above)
Comment (elefeint) [reply pom.xml]: The naming convention changed. I recall a conversation that R2DBC was going to release as 0.8 until the feature set solidified. So, there would be an 0.8 GA.

Here is the change on SPI side: https://github.com/r2dbc/r2dbc-spi/commit/e178e59ceaf27f55649ce684c97f1145fe8f3f06

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRowMetadata.java (hunk 2 above)
Comment (elefeint) [reply src/main/java/com/google/cloud/spanner/r2dbc/SpannerRowMetadata.java]: Nothing; I'll put it back. I default to field initialization when there is no specific reason for initializing in constructor, but there is no particular reason for it. I can see the argument for being consistent, and initializing everything in the constructor, too.

**Pull request merged**